src/suspendplugin.c

```diff
@@ -266,6 +276,40 @@ suspend_handle_display_on_off_iface(
     }
 
     nlmsg_free(msg);
+
+    char cmd[512];
+    int cmd_len = 0;
+    struct ifreq ifr;
+    android_wifi_priv_cmd priv_cmd;
+
+    int ret;
+    int ioctl_sock;
+
+    ioctl_sock = socket(PF_INET, SOCK_DGRAM, 0);
+
+    snprintf(cmd, 512, "SETSUSPENDMODE %d", on_off);
+    cmd_len = strlen(cmd);
+
+    memset(&ifr, 0, sizeof(ifr));
+    memset(&priv_cmd, 0, sizeof(priv_cmd));
+    strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
+
+    if (cmd_len > PRIV_CMD_SIZE) {
+        cmd_len = PRIV_CMD_SIZE - 1;
+    }
+
+    memcpy(priv_cmd.buf, cmd, cmd_len + 1);
```

Comment:
Why is this intermediate `cmd` buffer needed in the first place, why not to `snprintf` directly to `priv_cmd.buf`?